### ISSUE TITLE
make install or opam

### ISSUE THREAD

**Issue by lastland** [PrincetonUniversity/VST#151]
It would be helpful if VST can provide a `make install` option.

We are aware that there is `.loadpath-export` file, but we have several people collaborating on some VST files and everyone has a different VST load path. It would be great if we can just install VST in a common Coq load path, like `~/.opam/4.05.0/lib`? (I do not know if there are any other better options)

**Comment by lastland** [comment]
Has anyone tried to bother them to install those files in the "right place"?

**Comment by mansky1** [comment]
@ildyria Very cool! How close is this to a pull request allowing VST to be installed via opam by everyone?

**Comment by lastland** [comment]
I have just created a PR: https://github.com/coq/opam-coq-archive/pull/292

This should make CompCert install to the right place.

**Comment by andrew-appel** [comment]
The set of authors that @ildyria listed is not quite right.  It should be something like,

authors: ["Andrew W. Appel"
          "Lennart Beringer"
          "Sandrine Blazy"
          "Qinxiang Cao"
          "Santiago Cuellar"
          "Robert Dockins"
          "Josiah Dodds"
          "Nick Giannarakis"
          "Samuel Gruetter"
          "Aquinas Hobor"
          "Jean-Marie Madiot"
         ]

**Comment by lastland** [comment]
The abovementioned PR has been merged. CompCert should now install the `.vo` files at `.opam/<switch>/lib/coq/user-contrib/compcert`.

@ildyria Do you want to submit your opam file to [opam](https://github.com/coq/opam-coq-archive) so everyone can use it?

**Comment by ildyria** [comment]
@lastland while you would hope it does. It's not the case. Compcert is only installing a few files:

```
benoit@sandor:coq/user-contrib/compcert
▶ l
total 24K
drwxr-xr-x 3 benoit benoit 4.0K Apr 16 11:02 .
drwxr-xr-x 3 benoit benoit 4.0K Apr 16 11:02 ..
drwxr-xr-x 2 benoit benoit 4.0K Apr 16 11:02 include
-rw-r--r-- 1 benoit benoit  12K Apr 16 11:02 libcompcert.a

benoit@sandor:user-contrib/compcert/include
▶ l
total 36K
drwxr-xr-x 2 benoit benoit 4.0K Apr 16 11:02 .
drwxr-xr-x 3 benoit benoit 4.0K Apr 16 11:02 ..
-rw-r--r-- 1 benoit benoit 3.0K Apr 16 11:02 float.h
-rw-r--r-- 1 benoit benoit 1.9K Apr 16 11:02 stdalign.h
-rw-r--r-- 1 benoit benoit 2.5K Apr 16 11:02 stdarg.h
-rw-r--r-- 1 benoit benoit 1.8K Apr 16 11:02 stdbool.h
-rw-r--r-- 1 benoit benoit 3.5K Apr 16 11:02 stddef.h
-rw-r--r-- 1 benoit benoit 1.8K Apr 16 11:02 stdnoreturn.h
-rw-r--r-- 1 benoit benoit 1.9K Apr 16 11:02 varargs.h
```

I opened an issue and a PR with that respect:
https://github.com/AbsInt/CompCert/issues/227
https://github.com/AbsInt/CompCert/pull/228

**Comment by ildyria** [comment]
I submitted a pull request for opam:

https://github.com/coq/opam-coq-archive/pull/295

**Comment by lennartberinger** [comment]
Gordon Stewart's name is missing from the list of contributors.

**Comment by andrew-appel** [comment]
This issue should be resolved by pull request #406.
